### PR TITLE
add tinymidi/cci.20130325

### PR DESCRIPTION
**Pull request by ericLemanissier**
this is a port of https://github.com/bincrafters/conan-tinymidi

Specify library name and version:  **tinymidi/cci.20130325**

It's a dependency of sdl2_mixer

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
@madebr When I compile with system libtool, the build passes, but if I build_require libtool/2.4.6, it fails with 
```
libtool: compile: unable to infer tagged configuration
libtool:   error: specify a tag with '--tag'
```
Do you have an idea ?

File: recipes/tinymidi/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            raise ConanInvalidConfiguration("Only Linux is supported")
```
Comment (Croydon) [inline recipes/tinymidi/all/conanfile.py:28]:
```suggestion
            raise ConanInvalidConfiguration("Only Linux and FreeBSD is supported")
```

**Comment by madebr** [comment]
It's because of this: https://github.com/crosstool-ng/crosstool-ng/issues/940#issuecomment-381302380
tinymidy's `Makefile` sets `CC=gcc` which is different than the environment variable `CC=clang`.
https://github.com/krgn/tinymidi/blob/3162cf8faff04e26a8daa846618b90326f71b9d5/Makefile#L20
libtool detected an error: gcc vs clang

I pushed a fix to bincrafters' repo.

File: recipes/tinymidi/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/tinymidi/all/conanfile.py:39]: *are* supported :wink:

File: recipes/tinymidi/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        ]
+        if tools.get_env("CC"):
+            args.append("CC={}".format(tools.get_env("CC")))
+        return args
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            autotools = self._get_autotools()
+            make_args = self._make_args(autotools)
+            self.run("pwd")
```
Comment (madebr) [inline recipes/tinymidi/all/conanfile.py:69]:
```suggestion
```

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**